src/rust/engine/graph/src/lib.rs

```diff
@@ -633,15 +633,17 @@ impl<N: Node> Graph<N> {
       entry.cleaning_failed()
     }
 
-    // Otherwise, clear the deps.
-    // NB: Because `remove_edge` changes EdgeIndex values, we remove edges one at a time.
-    while let Some(dep_edge) = inner
+    // Otherwise, clear the deps. We remove edges in reverse index order, because `remove_edge` is
```

Comment:
This still is awfully titchy. I suppose though that retain_edges is too slow since it iterates every edge index? It otherwise hides all this same titch behind the API.

Reply:
Yea, `retain_edges` wouldn't allow us to actually use the edge indexes... `edges_directed` iterates over a lot fewer entries.